File: app/models/system_settings.py
# app/models/system_settings.py
from sqlalchemy import String
from sqlalchemy.orm import Mapped, mapped_column
from app.database import Base

class SystemSettings(Base):
    __tablename__ = "system_settings"

    id: Mapped[int] = mapped_column(primary_key=True, index=True)
    setting_key: Mapped[str] = mapped_column(String, unique=True, index=True, nullable=False)
    setting_value: Mapped[str] = mapped_column(String, nullable=False)
    descricao: Mapped[str | None] = mapped_column(String, nullable=True)

File: backend/internal/models/system_settings.go
package models

type SystemSetting struct {
	ID           uint    `gorm:"primaryKey" json:"id"`
	SettingKey   string  `gorm:"uniqueIndex;not null" json:"setting_key"`
	SettingValue string  `gorm:"not null" json:"setting_value"`
	Descricao    *string `json:"descricao"`
}

func (SystemSetting) TableName() string { return "system_settings" }
